tests/test_export.py

```diff
@@ -297,3 +300,34 @@ def test_bad_value_in_dep_mgmr_3(snippetcompiler):
         match="A dependency manager inserted a resource id without version this is not allowed aa::Bbbb\\[agent,name=agent\\]",
     ):
         snippetcompiler.do_export()
+
+
+def test_2121_wrapped_proxy_serialize(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+import exp
+
+dct = {"a": 1, "b": 2}
+
+x = exp::WrappedProxyTest(
+    name = "my_wrapped_proxy_test",
+    agent = "my_agent",
+    my_list = [1, 2, 3],
+    my_dict = {
+        "dct": dct,
```

Comment:
Best to also test that adding an entity/dynamicproxy inside a dict still causes an error on serialization.